cmake/O3DEJson.cmake

```diff
@@ -49,11 +49,35 @@ function(o3de_read_json_array read_output_array input_json_path array_key)
 endfunction()
 
 function(o3de_read_json_key output_value input_json_path key)
-    ly_file_read(${input_json_path} manifest_json_data)
+    ly_file_read_cache(${input_json_path} manifest_json_data)
     string(JSON value ERROR_VARIABLE manifest_json_error GET ${manifest_json_data} ${key})
     if(manifest_json_error)
         message(WARNING "Error reading field at key ${key} in file \"${input_json_path}\" : ${manifest_json_error}")
         return()
     endif()
     set(${output_value} ${value} PARENT_SCOPE)
 endfunction()
+
+#! o3de_read_json_keys: read multiple json keys at once. More efficient
+# than using o3de_read_json_key multiple times.
+# \arg:input_json_path - the path to the json file 
+# \args: pairs of 'key' and 'output_value'
+# e.g. To read the key1 and key2 values
+# o3de_read_json_keys(c:/myfile.json 'key1' out_key1_value 'key2' out_key2_value)
+function(o3de_read_json_keys input_json_path)
+    ly_file_read_cache(${input_json_path} manifest_json_data)
+    unset(key)
+    foreach(arg IN LISTS ARGN)
+        if(NOT key)
+            set(key ${arg})
+        else()
+            string(JSON value ERROR_VARIABLE manifest_json_error GET ${manifest_json_data} ${key})
+            if(manifest_json_error)
+                message(WARNING "Error reading field at key ${key} in file \"${input_json_path}\" : ${manifest_json_error}")
+            else()
+                set(${arg} ${value} PARENT_SCOPE)
+            endif()
+            unset(key)
+        endif()
+    endforeach()
```

Comment:
Uh, how are the keys being split from the value output parameter.

The function doc example indicates that the `key1` parameter is separate from the output parameter of `out_key1_value`, but this code is assuming the value parameter is empty.
If it is not, then it gets used as a key.

If anything we can use a separator to encode a single argument passed to this function as a key value.
Such as embeding a comma or newline in the parameter

Reply:
OK, I'll take a look at that change, though I may just remove this function for now as it is currently not used anywhere, it's from a previous iteration of the code and I thought it might be useful.

Reply:
Hmm, I tried to detect the issue when a value was not empty and would be used as a key, maybe I don't understand what the potential issues is yet, but in my test I didn't see any issue:
```
        unset(test_gem_name)
        unset(test_gem_display_name)

        # Grab the values when variables are empty
        o3de_read_json_keys("${gem_path}/gem.json" "gem_name" test_gem_name "display_name" test_gem_display_name)
        MESSAGE(INFO "gem_name: ${test_gem_name} display_name: ${test_gem_display_name}")

        # Grab the values when variables are not empty
        o3de_read_json_keys("${gem_path}/gem.json" "gem_name" test_gem_name "display_name" test_gem_display_name)
        MESSAGE(INFO "gem_name: ${test_gem_name} display_name: ${test_gem_display_name}")
   ```
   
   Prints out:
   ```
INFOgem_name: ArchVis display_name: Samples: ArchVis Assets
INFOgem_name: ArchVis display_name: Samples: ArchVis Assets
```

Reply:
Thanks! I'll make it use `if(NOT DEFINED` instead to avoid the "falsey" problem - which is also my new band name btw.